datadog/resource_datadog_synthetics_test_.go

```diff
@@ -2375,6 +2855,66 @@ func buildDatadogSyntheticsBrowserTest(d *schema.ResourceData) *datadogV1.Synthe
 	return syntheticsTest
 }
 
+func buildDatadogSyntheticsMobileTest(d *schema.ResourceData) *datadogV1.SyntheticsMobileTest {
+	syntheticsTest := datadogV1.NewSyntheticsMobileTestWithDefaults()
+	if attr, ok := d.GetOk("message"); ok {
+		syntheticsTest.SetMessage(attr.(string))
+	}
+	// if attr, ok := d.GetOk("monitor_id"); ok { // TODO uncomment and find out why it's unexpected
+	// 	syntheticsTest.SetMonitorId(int64(attr.(int)))
+	// }
+	if attr, ok := d.GetOk("name"); ok {
+		syntheticsTest.SetName(attr.(string))
+	}
+	if attr, ok := d.GetOk("status"); ok {
+		syntheticsTest.SetStatus(datadogV1.SyntheticsTestPauseStatus(attr.(string)))
+	}
+	if attr, ok := d.GetOk("type"); ok {
+		syntheticsTest.SetType(datadogV1.SyntheticsMobileTestType(attr.(string)))
+	}
+	config := datadogV1.SyntheticsMobileTestConfig{}
+	config.SetVariables([]datadogV1.SyntheticsConfigVariable{})
+
+	requestConfigVariables := d.Get("config_variable").([]interface{})
+	config.SetVariables(buildDatadogConfigVariables(requestConfigVariables))
+
+	if attr, ok := d.GetOk("config_initial_application_arguments"); ok {
+		initialApplicationArguments := attr.(map[string]interface{})
+		if len(initialApplicationArguments) > 0 {
+			config.SetInitialApplicationArguments(make(map[string]string))
+		}
+		for k, v := range initialApplicationArguments {
+			config.GetInitialApplicationArguments()[k] = v.(string)
+		}
+	}
+
+	syntheticsTest.SetConfig(config)
+
+	// if attr, ok := d.GetOk("device_ids"); ok { // TODO we don't want to build this because the response doesn't contain it and if you build it the local state goes all whack
```

Comment:
You mention the response and the state in this comment, but it's in the `buildDatadogSyntheticsMobileTest` function, which reads the state to build the request.

Reply:
If i remember correctly the issue here was that if put it in the request then in the end of the test it complained that this was not parsed in the response 🤔